### ISSUE TITLE
module deploy and redeploy not working

### ISSUE THREAD

**Issue by damonfeldman** [marklogic/marklogic-data-hub#1399]
#### The issue

Short description of the problem:
Some modules are not being loaded.

#### Tech details

Which Operating System are you using?  Mac OSX

Which version of MarkLogic are you using?  ML-9.0-6

Which version of the Data Hub Framework are you using? 4.0.0

Is this a QuickStart UI bug? No

#### The devil is in the details...

If possible, rerun the command with -d for debugging output and attach the output:

  **for quickstart:**  
  `java -jar quickstart.war -d > myoutput.txt`

  **for gradle:**  
  `gradle mlYourCommand -d > myoutput.txt`

It appears that the modules-timestamps file is not deleted, so files do not load. 
[reloadNoSMdepNoCode-debug.txt](https://github.com/marklogic/marklogic-data-hub/files/2456110/reloadNoSMdepNoCode-debug.txt)

**Comment by grechaw** [comment]
I often have to delete that timestamps file.  Is it that this should have happened on upgrade?  That would be a sensible thing to do.

**Comment by rjrudin** [comment]
I added this to fix the problem in a DHF4 project:

```
task deleteModuleTimestampsFile(type: Delete) {
	description = "Use this as mlDeleteModuleTimestampsFile doesn't seem to work with DHF4"
	delete "build/ml-javaclient-util/module-timestamps.properties"
}
mlReloadModules.dependsOn deleteModuleTimestampsFile
mlDeploy.dependsOn deleteModuleTimestampsFile
```

DHF has overwritten mlDeleteModuleTimestampsFile so that it instead does this:

```
@Override public String getUserModulesDeployTimestampFile() {
        return Paths.get(projectDir, ".tmp", USER_MODULES_DEPLOY_TIMESTAMPS_PROPERTIES).toString();
    }
```

Is there a need for a separate timestamps file? I would think the default one would work for all modules.

**Comment by ryanjdew** [comment]
This issue seems to be the root cause of an issue we see with DHF 4 and Smart Mastering.  +1 to the workaround by @rjrudin

**Comment by josvanroosmalen** [comment]
Please get this issue fixed. Unwind the ml-gradle fork.

**Comment by josvanroosmalen** [comment]
Related issue: #1574

**Comment by mhuang-ml** [comment]
@sbayatpur, This issue should have been resolved by release 4.1 and I believe can be closed.
